lib/harness/active_publisher.rb

```diff
@@ -21,8 +21,10 @@ module ActivePublisher
 
     ::ActiveSupport::Notifications.subscribe "message_published.active_publisher" do |*args|
       event = ::ActiveSupport::Notifications::Event.new(*args)
+      route = event.payload.fetch(:route, "")
+      route = ".#{route.gsub('.', '-')}" unless route.empty?
```

Comment:
Something like this would remove the string concat but honestly I think what you have is just fine.

```ruby
published_metric = route.empty? ? PUBLISHED_METRIC : "#{PUBLISHED_METRIC}.#{route.gsub('.', '-')}"

::Harness.count published_metric, message_count
```